torch/_inductor/scheduler.py

```diff
@@ -1330,7 +1342,39 @@ def compute_dependencies(self):
         Create dependency edges between nodes, handling aliasing and
         mutation properly.
         """
-        name_to_users: DefaultDict[str, List[NodeUser]] = collections.defaultdict(list)
+
+        T = TypeVar("T")
+
+        class DedupList(Generic[T]):
+            """
+            This data structure behaves like a list except it makes sure the
+            elements remain unique.
+            Normally one could use a set/dict for this purpose however
+            the list in question gets elements appended as it is being
+            iterated over which means that we need to keep the list
+            semantics.
```

Comment:
So `dict` is unique and order-preserving. Can't we use `dict` for this use case (i.e., adding while we enumerate)?

Reply:
Dict also do not allow for appending in the middle of iteration